runtime/lua/vim/ui.lua

```diff
@@ -32,5 +32,34 @@ function M.select(items, opts, on_choice)
   end
 end
 
+--- Prompts the user for input
+---
+---@param opts table Additional options. See |input()|
+---     - prompt (string|nil)
+---               Text of the prompt. Defaults to `Input: `.
+---     - default (string|nil)
+---               Default reply to the input
+---     - completion (string|nil)
+---               Specifies type of completion supported
+---               for input. Supported types are the same
+---               that can be supplied to a user-defined
+---               command using the "-complete=" argument.
+---               See |:command-completion|
+---     - highlight (function)
+---               Function that will be used for highlighting
+---               user inputs.
+---@param on_confirm function ((input) -> ())
+---               Called once the user confirms the input.
+---               `input` is what the user typed.
```

Comment:
Can we clarify what happens if the user aborts the input?

Reply:
Sure I can fix it. Right now it calls the function with the empty response from `vim.fn.input()`. Should we maybe make it so it only calls the `on_confirm()` if the input is not empty, or do we want it to be the responsibility of `on_confirm()` to handle this?

Reply:
I think it should be the responsibility of the handler, I would probably opt for a more generic name then on_confirm though

Reply:
I can change it to `handler`. I chose `on_confirm` to follow the naming of `vim.ui.select` though, since that one is called `on_choice`

Reply:
> I think it should be the responsibility of the handler

We need to make the contract clear so that both plugins utilizing `vim.ui.input` know what to expect and that plugins implementing a `vim.ui.input` override know what they need to do.

I'd suggest to use `nil` as indication that the user aborted. The argument could be called `on_input` - I think that would still fit despite the possibility to have a `nil` input.

An alternative would be to have a second `on_abort` callback.

Reply:
I changed it to use nil if the user aborted the input dialog. Let me know if there is anything else I should fix

Reply:
>I chose `on_confirm` to follow the naming of `vim.ui.select` though, since that one is called `on_choice`

Sticking with `on_` for callbacks is good as it's consistent with other areas in the Nvim API (`e.g. `on_key`), particularly if it's already used in `vim.ui.select`.

Reply:
Ok, so on_confirm vs on_input vs a second on_abort?

Reply:
Oh I actually thought @mfussenegger wrote `on_confirm` and not `on_input`. I would be happy to change it to something else than `on_confirm`, and for example `on_input`. But I do think two callbacks where one is for abort and one is for if confirmed is unnecessary. I think this could easily be handled in the callback that is already there by checking if the input is nil